src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/RazorSemanticTokenInfoServiceTest.cs

```diff
@@ -788,16 +880,88 @@ public async Task GetSemanticTokens_CSharp_ExplicitStatement_WithBackground()
             """;
 
         var razorRange = GetRange(documentText);
-        var csharpTokens = await GetCSharpSemanticTokensResponseAsync(documentText, razorRange, isRazorFile: true);
-        await AssertSemanticTokensAsync(documentText, isRazorFile: true, razorRange, csharpTokens: csharpTokens, withCSharpBackground: true);
+        var (csResponse, perRangeTokens) = await GetCSharpSemanticTokensResponseAsync(documentText, razorRange, isRazorFile: true);
+        await AssertSemanticTokensAsync(documentText, isRazorFile: true, razorRange, csResponse, perRangeTokens, withCSharpBackground: true);
+        Assert.NotNull(csResponse.Tokens);
+        Assert.NotEmpty(csResponse.Tokens);
+    }
+
+    [Fact]
+    public void StitchSemanticTokenResponsesTogether_OnNullInput_ReturnsEmptyResponseData()
```

Comment:
Should move these tests to their own file. We don't need them to run twice if they don't vary with the feature flag